codegen/codegen.go

```diff
@@ -300,15 +299,27 @@ func emitFilesystemSourceStmt(info *CodeGenInfo, scope *ast.Scope, call *ast.Cal
 			return st, err
 		}
 
-		id := identity.NewID()
-		info.Locals[id] = path
-
 		var opts []llb.LocalOption
 		for _, iopt := range iopts {
 			opt := iopt.(llb.LocalOption)
 			opts = append(opts, opt)
 		}
 
+		// get a consistent hash for this local (path + options) so we don't
+		// transport the same content multiple times when referenced repeatedly
+
+		// first get serialized bytes for this llb.Local state
+		tmpSt := llb.Local("constant-id-for-consistent-hashing", opts...)
```

Comment:
Could we just use `""`?

Reply:
yeah, in theory, I suspect `""` is an invalid value, and was concerned at some point in the future there might be a validation error when we try to marshal it.